tests/util.py

```diff
@@ -0,0 +1,97 @@
+import unittest
+import os
+import shutil
+from textwrap import dedent
+
+from mock import patch
+
+from docutils.io import StringOutput
+
+from sphinx.builders.text import TextBuilder
+from sphinx.application import Sphinx
+from sphinx.environment import SphinxFileInput
+
+
+class MockSphinx(Sphinx):
```

Comment:
We should probably unify how we're doing this everywhere, and break this into it's own package no PyPI. No sense in repeating it, and I imagine other folks would love to use it as well. The other option is to get it merged into Sphinx proper, as a real public API to do inmem testing.